core/google/api/core/gapic_v1/method.py

```diff
@@ -133,32 +131,32 @@ def __init__(self, target, retry, timeout, metadata):
 
     def __call__(self, *args, **kwargs):
         """Invoke the low-level RPC with retry, timeout, and metadata."""
-        # Note: Due to Python 2 lacking keyword-only arguments we kwargs to
+        # Note: Due to Python 2 lacking keyword-only arguments we use kwargs to
         # to extract the retry and timeout params.
         timeout_ = _determine_timeout(
             self._timeout,
-            kwargs.pop('timeout', None),
-            # Use the invocation-specified retry only for this, as we only
+            kwargs.pop('timeout', self._timeout),
+            # Use only the invocation-specified retry only for this, as we only
             # want to adjust the timeout deadline if the *user* specified
             # a different retry.
             kwargs.get('retry', None))
 
-        retry = kwargs.pop('retry', None)
-        if retry is None:
-            retry = self._retry
+        retry = kwargs.pop('retry', self._retry)
 
         # Apply all applicable decorators.
         wrapped_func = _apply_decorators(self._target, [retry, timeout_])
 
         # Set the metadata for the call using the metadata calculated by
         # _prepare_metadata.
-        if self._metadata is not False:
+        if self._metadata:
```

Comment:
ISTM that explicit `None` is a better idea here. What if someone uses `metadata = {}` but they want you to send it?

Reply:
Done.